### DISCUSSION TITLE
BLE version tends to crash after a few days - memory leak?

### DISCUSSION THREAD

**Discussion by EasilyBoredEngineer** [syssi/esphome-jk-bms#117] [General]
Hi @syssi,

I've just noted that the esp32 appears to crash (loses wifi / ble connection etc) after a few days using this repository. recovers after a power cycle. I am using the esp-idf framework, not the arduino.

Thought I'd mention in case anyone else seeing the same issue.

Cheers,

EBE

**Comment by syssi** [comment]
Is this crash reproducible/periodic? Does the setup crash later if you change the `throttle` parameter from `5s` to `10s`?

**Comment by syssi** [comment]
Did I understand correctly: The ESP doesn't recover from the crash automatically and you have to power cycle the device? I would expect a reboot.

**Reply by oakyz** [reply]
I’m having the same issue with 2 different esp32 boards, sometimes they work for days without issues sometimes they crash after 1 day. Both esp32 boards have wifi and bluetooth. They work fine after i reboot/power cycle them.

**Reply by syssi** [reply]
Please reflash your device using ESPHome 2022.8.3 and try again.

**Reply by oakyz** [reply]
> Please reflash your device using ESPHome 2022.8.3 and try again.

[logs_jk-bms-ble1_compile.txt](https://github.com/syssi/esphome-jk-bms/files/9608341/logs_jk-bms-ble1_compile.txt)

**Reply by syssi** [reply]
```
esphome clean yourconfig.yaml; esphome run yourconfig.yaml
```

Should solve the build issue.

**Reply by oakyz** [reply]
> ```
> esphome clean yourconfig.yaml; esphome run yourconfig.yaml
> ```
> 
> Should solve the build issue.

Worked like a charm :) thanks, still trying to figure out why the esp board crashes.... this time it worked for 1 week, i think it might have something to do with my wifi mesh being a 2.4+5ghz on same ssid... i'll probably get another router to keep it just for 2.4ghz, curious thing is when the esp board crashes is disconnected from BMS and cannot access it even if it's connected to HA by USB cable.

**Reply by syssi** [reply]
My theory are WiFi reconnects or BLE reconnects which consumes much memory/resources temporary (and let the ESP crash) or memory leaks in case of reconnects. My BLE test setup is stable if I throttle the traffic to `5s` and don't try to use all supported sensors.
